parsers/VN.py

```diff
@@ -1,216 +1,167 @@
 #!/usr/bin/env python3
 
-from datetime import datetime
+import json
+import zoneinfo
+from datetime import datetime, timedelta
 from logging import Logger, getLogger
-from typing import Any
 
-import arrow
-import pytz
-from requests import Response, Session
+from requests import Session
 
+from electricitymap.contrib.config import ZoneKey
+from electricitymap.contrib.lib.models.event_lists import (
+    PriceList,
+    TotalConsumptionList,
+)
 from parsers.lib.exceptions import ParserException
 
 ## Vietnamese National Load Dispatch Center https://www.nldc.evn.vn/
 # Access via day, can also parse historical data
 
-tz = pytz.timezone("Asia/Ho_Chi_Minh")
-base_url_from_type = {
-    "consumption": "https://www.nldc.evn.vn/PhuTaiHandle.ashx?d=",  # + dd/mm/yyyy
-    "price": "https://www.nldc.evn.vn/GiaBienHandle.ashx?d=",  # + dd/mm/yyyy
+tz = zoneinfo.ZoneInfo("Asia/Ho_Chi_Minh")
+LIVE_DATA = {
+    "consumption": "https://www.nldc.evn.vn/api/services/app/Pages/GetChartPhuTaiVM",
+    "price": "https://www.nldc.evn.vn/api/services/app/Pages/GetChartGiaBienVM",
 }
 
+HISTORICAL_DATA = "https://www.nldc.evn.vn/api/services/app/Dashboard/GetBieuDoTuongQuanPT"  # + dd/mm/yyyy
 
-def fetch_latest_data(
-    data_type: str,
-    session: Session,
-):
-    """
-    Fetch latest available
-    """
-    fetched_day = arrow.now(tz=tz)
-
-    base_url = base_url_from_type[data_type]
-
-    today_dt_str = fetched_day.format("DD/MM/YYYY")
-
-    # Try fetch data for today.
-    today_response: Response = session.get(base_url + today_dt_str)
-
-    # Check if today's data is there.
-    if today_response.ok and today_response.json() is not None:
-        return fetched_day.floor("day"), today_response.json()
-
-    # If it is too early in the day, it might not contain data yet. Get from yesterday.
-    fetched_day = fetched_day.shift(days=-1)
-    yesterday_dt_str = fetched_day.format("DD/MM/YYYY")
-
-    yesterday_response: Response = session.get(base_url + yesterday_dt_str)
-    # Check if yesterday's data is there.
-    if yesterday_response.ok and yesterday_response.json() is not None:
-        return fetched_day.floor("day"), yesterday_response.json()
-
-    # Also no data from yesterday, cant fetch recent data.
-    raise ParserException(
-        parser="VN.py",
-        message=(
-            f"Requested recent data - request endpoint can't provide data for 2 days."
-            f"Return codes: {today_response.status_code}, {yesterday_response.status_code} at {fetched_day}"
-        ),
-    )
-
+ZONE_CODE = {ZoneKey("VN"): "HT"}
 
-def fetch_target_data(data_type: str, target_datetime, session: Session):
-    """
-    Fetch the day containing the given target_datetime.
-    """
 
-    # Convert timezone, and get the day
-    target_datetime_vn = arrow.get(target_datetime).to(tz)
+def fetch_consumption(
+    zone_key: ZoneKey,
+    session: Session = Session(),
```

Comment:
```suggestion
    session: Session | None,
```
Due to python weirdness we should not set defualt object like this.

it should be set just below as:
`session = session or Session()`

That goes for all the top level functions, then we should just pass on the session.